experimental/source/source.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2020-2024 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// package source provides standard queries and interfaces for accessing the
+// contents of source files.
+package source
+
+import (
+	"errors"
+	"io"
+	"io/fs"
+	"strings"
+
+	"github.com/bufbuild/protocompile/experimental/incremental"
+	"github.com/bufbuild/protocompile/experimental/report"
+)
+
+// Stage is the value this package uses for [report.Report].Stage.
+const Stage int = 0
```

Comment:
What exactly is this meant to represent? Might it be better as a string to describe the stage? What are other stages and what is the significance of their numeric value?

Reply:
This was a poorly-named variable. I've done some refactoring (including moving the query itself into a new dedicated packge). LMK what you think.